### PR TITLE
TDSLoader: parse opacity of material

### PR DESCRIPTION
**Pull request by wemamawe**
some model is transparent, so it should parse MAT_TRANSPARENCY, and convert MAT_TRANSPARENCY to material.oapcity

### CODE REVIEW

File: examples/js/loaders/TDSLoader.js [hunk 1]
```diff
@@ -272,6 +272,12 @@ THREE.TDSLoader.prototype = Object.assign( Object.create( THREE.Loader.prototype
 				material.shininess = shininess;
 				this.debugMessage( '   Shininess : ' + shininess );
 
+			} else if ( next === MAT_TRANSPARENCY ) {
```
Comment (WestLangley) [inline examples/js/loaders/TDSLoader.js]: Did you test this PR?

Where is `MAT_TRANSPARENCY` defined?

File: examples/js/loaders/TDSLoader.js [hunk 2]
```diff
@@ -272,6 +272,12 @@ THREE.TDSLoader.prototype = Object.assign( Object.create( THREE.Loader.prototype
...
+
+				var opacity = this.readWord( data );
+				material.opacity = opacity*0.01;
+				this.debugMessage( '  Opacity : ' + opacity );
+                		material.transparent = opacity<100 ? true : false;
```
Comment (WestLangley) [inline examples/js/loaders/TDSLoader.js]: Formatting
```js
var opacity = this.readWord( data );
material.opacity = opacity * 0.01;
this.debugMessage( '  Opacity : ' + opacity );
material.transparent = opacity < 100 ? true : false;
```
Please remember to use tabs, not spaces.

File: examples/js/loaders/TDSLoader.js (hunk 1 above)
Comment (wemamawe) [reply examples/js/loaders/TDSLoader.js]: yes, i had tested, first i read code of lib3ds 
`case CHK_MAT_SHININESS: {
                lib3ds_chunk_read_reset(&c, io);
                int_percentage_read(&material->shininess, io);
                break;
            }

            case CHK_MAT_SHIN2PCT: {
                lib3ds_chunk_read_reset(&c, io);
                int_percentage_read(&material->shin_strength, io);
                break;
            }

            case CHK_MAT_TRANSPARENCY: {
                lib3ds_chunk_read_reset(&c, io);
                int_percentage_read(&material->transparency, io);
                break;
            }
`
then, There also has MAT_TRANSPARENCY parameter in TDSLoader

Comment (WestLangley) [reply examples/js/loaders/TDSLoader.js]: `MAT_TRANSPARENCY` is undefined, so your code will throw an error.

Comment (wemamawe) [reply examples/js/loaders/TDSLoader.js]: `
var MAT_TRANSPARENCY = 0xA050;
`
it is defined in TDSLoader.js  at line 965

Comment (mrdoob) [reply examples/js/loaders/TDSLoader.js]: The line is commented out:
https://github.com/mrdoob/three.js/blob/c52fef3b548ecf7d252850ffaa9a2eaca6b26989/examples/js/loaders/TDSLoader.js#L921

Comment (wemamawe) [reply examples/js/loaders/TDSLoader.js]: oh, the version of my local THREE.js is v101, that line didn't commented

Comment (mrdoob) [reply examples/js/loaders/TDSLoader.js]: Better use the `dev` branch as base for your changes when submitting PRs.

Comment (wemamawe) [reply examples/js/loaders/TDSLoader.js]: yes, it is > dev branch in my github account

Comment (mrdoob) [reply examples/js/loaders/TDSLoader.js]: You just said that you used r101 as base for your PR. `dev` is on r109dev...

Comment (wemamawe) [reply examples/js/loaders/TDSLoader.js]: sorry, maybe i didn't say very clearly, In my github account, i fllow your `dev` version, and at my local computer, i used 101 version.

File: examples/js/loaders/TDSLoader.js (hunk 2 above)
Comment (Mugen87) [reply examples/js/loaders/TDSLoader.js]: @wemamawe Still interested in fixing this formatting issues? Besides, the loader is still missing the declaration of `MAT_TRANSPARENCY  and thus produces a runtime error.

Live demo: https://jsfiddle.net/7ju6Lw28/

Comment (wemamawe) [reply examples/js/loaders/TDSLoader.js]: @Mugen87 I delete that coment befor `var MAT_TRANSPARENCY = 0xA050;`

Comment (Mugen87) [reply examples/js/loaders/TDSLoader.js]: > I delete that coment befor var MAT_TRANSPARENCY = 0xA050;

Have you forgotten to commit this change? I can't see this change in the diff (besides, the live demo is still broken).

**Comment by mrdoob** [comment]
@wemamawe here you can see the changes in this PR:

https://github.com/mrdoob/three.js/pull/17456/files

As you can see, it's still missing the `MAT_TRANSPARENCY` declaration.

File: examples/js/loaders/TDSLoader.js (hunk 2 above)
Comment (wemamawe) [reply examples/js/loaders/TDSLoader.js]: image
now it works

**Comment by Mugen87** [comment]
Resolved style issues and updated module here: #17534

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**